src/vs/workbench/contrib/terminal/browser/terminalActions.ts

```diff
@@ -1470,36 +1468,31 @@ export function registerTerminalActions() {
 			if (customType) {
 				return commandService.executeCommand(customType.command);
 			}
+			terminalService.configHelper.setWorkspaceShellAllowed(false);
 			if (detectedProfiles) {
-				let launchConfig = detectedProfiles?.find((profile: { profileName: string; }) => profile.profileName === profileSelection);
-				if (launchConfig && !launchConfig.isWorkspaceProfile) {
-					const instance = terminalService.createTerminal({ executable: launchConfig.path, args: launchConfig.args });
-					terminalService.setActiveInstance(instance);
-				} else if (launchConfig && launchConfig.isWorkspaceProfile) {
-					if (!storageService.getBoolean(`${QUICK_LAUNCH_PROFILE_CHOICE}-${item}-${launchConfig.path}`, StorageScope.WORKSPACE, false)) {
-						notificationService.prompt(Severity.Info, `Do you allow this workspace to modify your terminal profile? ${item} with path ${launchConfig?.path}`,
+				const launchConfig = detectedProfiles?.find((profile: { profileName: string; }) => profile.profileName === profileSelection);
+				if (launchConfig) {
+					if (!terminalService.configHelper.checkWorkspaceShellPermissions() || !launchConfig.isWorkspaceProfile) {
```

Comment:
I was about to say we want to check `isWorkspaceProfile` and fall through first before attempting to call `checkWorkspaceShellPermissions` since it triggers a notification, but we don't need to check it at all. `checkWorkspaceShellPermissions` will check not only the shell and args for us but also if `env` is set which is also important.